PowerEditor/src/Utf8_16.cpp

```diff
@@ -442,6 +443,12 @@ void Utf8_16_Write::fclose()
 
 	if (m_pFile)
 	{
+		if (updateEOF)
+		{
+			const int fd = _fileno(m_pFile);
+			_chsize_s(fd, _ftelli64(m_pFile));
+		}

```

Comment:
Why do we need reset the file length here?
Does `fflush `not put what we need exactly?

Reply:
When file is opened with "rb+" its content is not erased and the EOF (end-of-file) mark is not updated as well.
We begin to write the new data over the previous and if the file grows bigger the EOF mark is updated properly (file is enlarged).
On the other hand, if the written new data is less and the file is supposed to shrink the EOF mark is not automatically updated (because the file is opened for read/write access and the remaining old data we had not updated is kept intact).
So if we want to shrink the file we have to explicitly truncate it to the end of the new written content. `_ftelli64()` should tell us the current position in the file since the last successful write and we move the EOF there truncating the unneeded content.
Unfortunately I couldn't find standard CRT function that directly sets the EOF mark position and thus I had to use `_chsize_s()`

Reply:
So are these 2 lines for shrinking the closing file ?

Reply:
Yes